docs/installation/getting-started/editor-lifecycle.md

```diff
@@ -3,167 +3,157 @@ category: getting-started
-</div>
-```
+After creating the editor, you might want to use the editor's API. But first, you need to the editor's instance.
 
-Then call {@link module:editor-inline/inlineeditor~InlineEditor#create `InlineEditor.create()`} to **attach** {@link installation/getting-started/predefined-builds#inline-editor inline editor} to the `<div>` element:
+### Acquiring the editor's instance
+
+The simplest way would be to save the reference to the editor somewhere after you create it. This is often done by using a window or some state management object. You will often see lines like this in our docs.
 
 ```js
-InlineEditor
-	.create( document.querySelector( '#editor' ) )
-	.then( editor => {
-		console.log( editor );
-	} )
-	.catch( error => {
-		console.error( error );
-	} );
+// Editor's creation steps.
+// ...
+.then((editor) => {
+	window.editor = editor;
+})
 ```
 
-### Example – balloon editor
-
-The procedure is the same as for the inline editor. The only difference is that you need to use the {@link module:editor-balloon/ballooneditor~BalloonEditor#create `BalloonEditor.create()`} method.
+The second option is to make a simple function plugin that will be added to the plugins list. The first parameter of this function will be the editor's instance.
 
-Add an element where CKEditor should initialize to your page:
-
-```html
-<div id="editor">
-	<p>Here goes the initial content of the editor.</p>
-</div>
+```js
+function myPlugin(editor) {
+	// Interact with the API.
+	// ...
+}
+
+ClassicEditor.create(document.querySelector("#editor"), {
+	// If you're using builds, this is going to be extraPlugins property.
+	plugins: [
+		myPlugin,
+		// Other plugins.
+		// ...
+	],
+});
 ```
 
-Then call {@link module:editor-balloon/ballooneditor~BalloonEditor#create `BalloonEditor.create()`} to **attach** {@link installation/getting-started/predefined-builds#balloon-editor balloon editor} to the `<div>` element:
+This method allows writing simple plugins that will be executed during the initialization of the editor, and don't need to interact with other plugin schema's or UI.
+
+The last option is to make a plugin that inherits from the Plugin class:
 
 ```js
-BalloonEditor
-	.create( document.querySelector( '#editor' ) )
-	.then( editor => {
-		console.log( editor );
-	} )
-	.catch( error => {
-		console.error( error );
-	} );
+class MyPlugin extends Plugin {
+	init() {
+		const editor = this.editor;
+		// Interact with the API.
+		// ...
+	}
+}
+
+ClassicEditor.create(document.querySelector("#editor"), {
+	// If you're using builds, this is going to be extraPlugins property.
+	plugins: [
+		MyPlugin,
+		// Other plugins.
+		// ...
+	],
+});
 ```
 
-### Example – decoupled editor
+Plugins created in this way can do things after initialization (afterInit) as well as set up new editor UI components.
+TODO links to plugin interface and framework.
 
-Add the elements where CKEditor should initialize the toolbar and the editable to your page:
+### Using the API
```

Comment:
I see this as a separate category - API and Events. We could leave the "editors lifecycle" as is then. We could split examples into sub-articles for better findability. Explain what command is, what writer is, and how to use events.